lines/client/rundown.py

```diff
@@ -1,57 +1,89 @@
+_LOGGER = logging.getLogger(__name__)
+
+
+class Rundown(Task):
+    """
+    TheRundown API Wrapper Task.
 
+    Rundown is a purely designed API wrapper for TheRundown.io REST API
+    The task will be mapped to fetch lines for many sports
 
-class RundownClient(object):
-    """TheRundown API Wrapper Client."""
+    This will be carried out by interval schedule and mapped according.
 
-    def __init__(self):
-        """Constructor for RundownClient."""
-        self.base_url = "https://therundown-therundown-v1.p.rapidapi.com/"
-        self._api_host = os.getenv("RUNDOWN_API_HOST")
-        self._api_key = os.getenv("RUNDOWN_API_KEY")
-        self._today = datetime.datetime.now()
+    Args:
+        sport: (int) - sport_id value of `RundownSportId`
+    Returns:
+        lines: (LinesResponseFacade) - facade of response object `.json()`
+    """
 
-    def __repr__(self):
-        """Representation for RundownClient."""
-        return "<RundownClient: {}>".format(str(self._today))
+    def __init__(self, *args, **kwargs):
+        """Constructor for Rundown"""
+        self.base_url = "https://therundown-therundown-v1.p.rapidapi.com"
+        self.rundown_host = os.getenv("RUNDOWN_HOST")
+        self.rundown_key = os.getenv("RUNDOWN_KEY")
+        super().__init__(*args, **kwargs)
 
-    def lines(self, sport_id: int):
+    def run(self, sport: int) -> LinesResponseFacade:
         """
         Fetch live lines via TheRundown.
 
         Args:
-            sport_id: (int) - SportId for TheRundown API request
+            sport: (int) - sportId integer for TheRundown request
         Returns:
-            response.json(): (dict) - response object jsonized
-            NOTE: raises for invalid SportId and HTTP status
+            lines: (LinesResponseFacade) - facade of response object `.json()`
+            NOTE: raises for InvalidSportId and `Response.raise_for_status()`
         """
-        if sport_id not in RundownSportId._value2member_map_:
-            raise ValueError("{} is not a valid SportId".format(sport_id))
+        if sport not in RundownSportId._value2member_map_:
+            raise InvalidSportIdError(
+                "Valid `sport` values: {}".format(
+                    RundownSportId.list()
+                )
+            )
 
-        url_extras = "sports/{}/events/{}".format(
-            str(sport_id), self._today.strftime("%Y-%m-%d")
+        now = datetime.datetime.now()
+        request_url = "{}/sports/{}/events/{}".format(
+            self.base_url, str(sport), now.strftime("%Y-%m-%d")
+        )
+
+        _LOGGER.info(
+            "<URL BUILT> {}".format(
+                request_url
+            )
         )
-        request_url = self.base_url + url_extras
 
         params = {
-            "include": ["all_periods", "scores"],
+            "include": [
+                "all_periods",
+                "scores"
+            ],
             "offset": "0"
         }
 
         headers = {
```

Comment:
Can be held in the `__init__`, your call.

Reply:
Would prefer to not include function/endpoint specific variables in __init__